### PR TITLE
Instant.now() removed from Kademlia

### PR DESCRIPTION
**Pull request by alari**

### CODE REVIEW

File: kademlia/core/src/main/scala/fluence/kad/Bucket.scala [hunk 1]
```diff
@@ -61,18 +67,25 @@ case class Bucket[C](maxSize: Int, nodes: Queue[Node[C]] = Queue.empty) {
...
    * @param node Node
    * @param pingExpiresIn Duration to ignore updates for a node
    */
-  def shouldUpdate(node: Node[C], pingExpiresIn: Duration): Boolean =
-    find(node.key).fold(true)(
-      n ⇒
-        !pingExpiresIn.isFinite() ||
-          java.time.Duration.between(n.lastSeen, node.lastSeen).toMillis >= pingExpiresIn.toMillis
-    )
+  def shouldUpdate(node: Node[C], pingExpiresIn: Duration, currentTime: Long): Boolean =
```
Comment (RicoGit) [inline kademlia/core/src/main/scala/fluence/kad/Bucket.scala]: There is no doc for currentTime.

File: kademlia/core/src/main/scala/fluence/kad/Bucket.scala [hunk 2]
```diff
@@ -94,55 +107,75 @@ object Bucket {
...
-
-            // Ping last contact.
-            // If it responds, enqueue it and drop the new node, otherwise, drop it and enqueue new one
-            StateT.liftF(rpc(last.contact).ping().attempt.to[F]).flatMap {
-              case Left(_) ⇒
-                StateT.set(b.copy(nodes = nodes.enqueue(node))).map(_ ⇒ true)
-              case Right(updatedLastContact) ⇒
-                StateT.set(b.copy(nodes = nodes.enqueue(updatedLastContact))).map(_ ⇒ false)
-            }
+          time.map(t ⇒ !pingExpiresIn.isFinite() || t - last.lastSeenEpochMillis <= pingExpiresIn.toMillis).flatMap {
```
Comment (RicoGit) [inline kademlia/core/src/main/scala/fluence/kad/Bucket.scala]: It's hard to understand, let's create function\method with clear name for `!pingExpiresIn.isFinite() || t - last.lastSeenEpochMillis <= pingExpiresIn.toMillis`

File: kademlia/core/src/main/scala/fluence/kad/RoutingTable.scala [hunk 3]
```diff
@@ -117,25 +117,24 @@ class RoutingTable[C](nodeId: Key)(implicit SR: Siblings.ReadOps[C], BR: Bucket.
...
-    else {
+    else
       checkNode(node).attempt.to[F].flatMap {
         case Right(true) ⇒
           logger.trace("Update node: {}", node.key)
-          for {
-            // Update bucket, performing ping if necessary
-            savedToBuckets ← BW.update((node.key |+| nodeId).zerosPrefixLen, node, rpc, pingExpiresIn)
 
+          P sequential P.apply.map2(
```
Comment (RicoGit) [inline kademlia/core/src/main/scala/fluence/kad/RoutingTable.scala]: Why style without point is better here?

File: kademlia/core/src/main/scala/fluence/kad/Bucket.scala [hunk 4]
```diff
@@ -30,18 +33,21 @@ import scala.language.higherKinds
 
 /**
  * Kademlia's K-Bucket
+ *
  * @param maxSize Max size of bucket, K in paper
- * @param nodes Queue of nodes; last one is the least recent seen
+ * @param records Queue of records; last one is the least recent seen
```
[the remark this answers is not in the document]
Comment (alari) [reply kademlia/core/src/main/scala/fluence/kad/Bucket.scala]: `records` is the name of the field, it must match. `Queue` is the first word of field docs, that's why it's capital cased.

**Pull request merged**